docs/installation.rst

```diff
@@ -0,0 +1,424 @@
+   +----------------------------------------+---------------------------------------------------------------------------------------------+
+
+
+.. note::
+
+  TensorFlow installation is required to build the TensorFlow plugin for DALI.
+
+.. note::
+
+  Items marked *"unofficial"* are community contributions that are believed to work but not officially tested or maintained by NVIDIA.
+
+.. note::
+
+   This software uses the FFmpeg licensed code under the LGPLv2.1. Its source can be downloaded `from here. <https://developer.download.nvidia.com/compute/redist/nvidia-dali/ffmpeg-3.4.2.tar.bz2>`_
+
+   FFmpeg was compiled using the following command line:
+
+.. code-block:: bash
+
+    ./configure \
+     --prefix=/usr/local \
+     --disable-static \
+     --disable-all \
+     --disable-autodetect \
+     --disable-iconv \
+     --enable-shared \
+     --enable-avformat \
+     --enable-avcodec \
+     --enable-avfilter \
+     --enable-protocol=file \
+     --enable-demuxer=mov,matroska \
+     --enable-bsf=h264_mp4toannexb,hevc_mp4toannexb && \
+     make
+
+
+
+Get the DALI source
+^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+    git clone --recursive https://github.com/NVIDIA/dali
+    cd dali
+
+Make the build directory
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+    mkdir build
+    cd build
+
+
+Compile DALI
+^^^^^^^^^^^^
+
+Building DALI without LMDB support:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+    cmake ..
+    make -j"$(nproc)"
+
+
+Building DALI with LMDB support:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+    cmake -DBUILD_LMDB=ON ..
+    make -j"$(nproc)"
+
+
+Building DALI using Clang (experimental):
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+
+   This build is experimental. It is neither maintained nor tested. It is not guaranteed to work.
+   We recommend using GCC for production builds.
+
+
+.. code-block:: bash
+
+    cmake -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang  ..
+    make -j"$(nproc)"
+
+
+**Optional CMake build parameters**:
+
+-  ``BUILD_PYTHON`` - build Python bindings (default: ON)
+-  ``BUILD_TEST`` - include building test suite (default: ON)
+-  ``BUILD_BENCHMARK`` - include building benchmarks (default: ON)
+-  ``BUILD_LMDB`` - build with support for LMDB (default: OFF)
+-  ``BUILD_NVTX`` - build with NVTX profiling enabled (default: OFF)
+-  ``BUILD_TENSORFLOW`` - build TensorFlow plugin (default: OFF)
+-  ``BUILD_NVJPEG`` - build with ``nvJPEG`` support (default: ON)
+-  ``BUILD_NVOF`` - build with ``NVIDIA OPTICAL FLOW SDK`` support (default: ON)
+-  ``BUILD_NVDEC`` - build with ``NVIDIA NVDEC`` support (default: ON)
+-  ``BUILD_NVML`` - build with ``NVIDIA Management Library`` (``NVML``) support (default: ON)
+-  ``WERROR`` - treat all build warnings as errors (default: OFF)
+-  ``BUILD_WITH_ASAN`` - build with ASAN support (default: OFF). To run issue: LD_LIBRARY_PATH=.
```

Comment:
The instruction how to run is a little bit cluttered (at least in GH render). Maybe move it to separate paragraph?